enhancements/ingress/gateway-api-crd-life-cycle-management.md

```diff
@@ -0,0 +1,467 @@
+the admin gate.
+
+The admin will be responsible for ensuring the safety of succession. If they
+force through the pre-upgrade check AND the admin gate, the admin gate will be
+left behind to help aid investigation of the cause of problems when the upgrade
+goes wrong.
+
+[an effort]:https://github.com/kubernetes-sigs/gateway-api/discussions/3497
+
+### Workflow Description
+
+The workflow in this case is an upgrade process. From the _user_ perspective the
+CRDs will be fully managed via the platform from here on out, so they only need
+to interface with the upgrade workflow on the condition that their cluster had
+previously installed Gateway API CRDs. The workflow consists of the pre-upgrade
+checks and the post-upgrade checks.
+
+### Pre-upgrade
+
+1. In the CIO a pre-upgrade check verifies CRD presence
+  * IF the CRDs are present
+    * an admingate is created requiring acknowledgement of CRD succession
+    * UNTIL the schema matches an exact versions we set `Upgradable=false`
+2. Once CRDs are not present OR an exact match we set `Upgradable=true`
+
+> **Note**: A **cluster-admin** is required for these steps.
+
+> **Note**: The logic for these lives in in the previous release (4.18), but
+> does not need to be carried forward to future releases as other logic exists
+> there to handle Gateway API CRD state (see below).
+
+### Post-upgrade
+
+> **Note**: The logic for these lives in in the new release (4.19) and onward.
+
+1. The CIO is hereafter deployed alongside its CRD protection VAP
+2. The CIO constantly checks for the presence of the CRDs
+  * IF the CRDs are present
+    * UNTIL the CRD schema matches what is expected, the CIO upgrades them
+    * IF the upgrade fails `Degraded` status is set
+  * ELSE the CRDs are deployed by the CIO
+
+> **Note**: If we reach `Degraded` its expected that some tampering has
+> occurred (e.g. a cluster-admin has for some reason destroyed our VAP and
+> manually changed the CRDs). For the initial release we will simply require
+> manual intervention (support) to fix this as we can't guess too well at the
+> original intent behind the change. In future iterations we may consider more
+> solutions if this becomes a common problem for some reason.
+
+### API Extensions
+
+None.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+Hypershift runs the Ingress Operator on the management cluster but configured
+with a kubeconfig to manage resources on the guest cluster.  This means that the
+Ingress Operator manages the Gateway API CRDs on the guest cluster, the same as
+on standalone clusters.
+
+#### Standalone Clusters
+
+For standalone clusters, the Ingress Operator manages the Gateway API CRD
+life-cycle so that Gateway API can be configured and used post-install.
+
+#### Single-node Deployments or MicroShift
+
+The CRDs themselves use minimal resources.  Creating a GatewayClass CR can cause
+the Ingress Operator to install OpenShift Service Mesh, which in turn installs
```

Comment:
You could create a VAP set to WARN that would notify a user that their action implicitly enables service mesh, and give them some feedback there, might be a useful addition

Reply:
Interestingly we _don't_ enable service mesh. It's kinda weird, but we're literally just using Istio for it's ingress Gateway capabilities (i.e. we don't deploy `istio-cni`, or allow any use of service mesh features, etc). We're wanting more distinctive support for this use case in the future, which the OSSM team has been helping us with. For the deployment of OSSM we'll be doing, the Gateway API resources are the _only_ API surface we will be supporting (in fact [we want to soft block someone trying going around this](https://issues.redhat.com/browse/NE-1966)). All that said, we recently split the feature gates for the gateway API CRDs and the OSSM implementation, so OSSM is no longer a part of this and the EP was a bit out of date. This is now updated to remove references to OSSM, LMKWYT

Reply:
We intend to make the service mesh components of OSSM inaccessible in this deployment, could you help me to better understand what you envision this warning saying, and what explicit action would trigger it?